src/argilla/server/apis/v1/handlers/oauth2.py

```diff
@@ -61,13 +66,14 @@ async def get_access_token(request: Request, provider: str, db: AsyncSession = D
 
         user = await accounts.get_user_by_username(db, username)
         if user is None:
-            await accounts.create_user_with_random_password(
+            user = await accounts.create_user_with_random_password(
                 db,
                 username=username,
                 first_name=user_info.name,
                 role=_USER_ROLE_ON_CREATION,
                 workspaces=[workspace.name for workspace in settings.oauth.allowed_workspaces],
             )
+            telemetry.track_user_created(user, is_oauth=True)
```

Comment:
If the user is not created for some reason we are sending telemetry data (empty I guess)?. If that's the case maybe It makes sense to move the telemetry call to be inside `accounts.create_user_with_random_password` function.

Reply:
If the user is because the function `accounts.create_user_with_random_password` fails, so the next line won't be executed.

The user-created event will raised only the the user was properly created